pkg/client/client.go

```diff
@@ -1245,6 +1244,52 @@ func (c *Client) WaitForSecretByNsName(ctx context.Context, obj types.Namespaced
 	return c.WaitForSecret(ctx, &secret)
 }
 
+func (c *Client) WaitForConfigMap(ctx context.Context, cm *v1.ConfigMap) (*v1.ConfigMap, error) {
```

Comment:
nit: Is this code not used? Or am I missing something 🤔 Do we have plans to start using it?

Reply:
good catch! I forgot to modify the Prometheus adapter and control plane tasks.